python/tests/test_material_grid.py

```diff
@@ -18,8 +18,8 @@ def compute_resonant_mode(res,default_mat=False):
 
         design_shape = mp.Vector3(1,1,0)
         design_region_resolution = 50
-        Nx = int(design_region_resolution*design_shape.x)
-        Ny = int(design_region_resolution*design_shape.y)
+        Nx = int(design_region_resolution*design_shape.x)+1
```

Comment:
Why are we adding 1 here? I realize it "fixes" a failing test, but this doesn't make sense to me.

Reply:
The first and last pixels of the material grid lie on the boundaries of the grid — that means the number of "pixels" in that direction is actually N-1 where N is the number of grid points.

Reply:
A `desigion_region_resolution` of `50` and `design_shape` of `(1,1,0`) correspond to a Δx and Δx of `0.2`. Without adding one to `Nx` and `Ny` (the current setup),  Δx and Δx are `0.020408` which is slightly off. Another solution is to use `x = np.linspace(-0.5*design_shape.x,0.5*design_shape.x,Nx,endpoint=False)`, etc.

Reply:
Ah yes that makes sense, thanks!